transform/mattermost-analytics/models/intermediate/product/active_users/int_user_active_days_legacy_telemetry.sql

```diff
@@ -0,0 +1,45 @@
+{{
+    config({
+        "materialized": "incremental",
+        "incremental_strategy": "merge",
+        "merge_update_columns": ['received_at_date'],
+        "unique_key": ['daily_user_id'],
+        "cluster_by": ['received_at_date'],
+        "snowflake_warehouse": "transform_l"
+    })
+}}
+
+with tmp as (
+    select
+        cast(received_at as date) as received_at_date,
+        cast(timestamp as date) as activity_date,
+        server_id,
+        user_id
+    from
+        {{ ref('stg_mattermost2__tracks') }}
+    where
+        -- Exclude items without user ids, such as server side telemetry etc
+        user_id is not null
+        -- Exclude items without server ids
+        and server_id is not null
```

Comment:
Noticed that user_id being not NULL (renamed here as server_id) does not guarantee that these are analytics.js events only, there are cases where both analytics.js and analytics-go libraries generate such events.

Reply:
As far as I understand, `server_id` (coming from `user_id`) must not be `null`, as it's expected to be there to denote the server that the event is originating from. If it is equal to null, then it's some special case that needs to be investigate.

Regarding `user_id` (coming from `user_actual_id`), the goal was to keep only user generated events, and not server generated. I agree that this can be further optimized to exclude such events earlier in the pipeline. However, I think it's a good point for the second iteration. The reasons are:
- Current full refresh process took 1 hour. Daily refreshes should take way less.
- The optimization would require excluding specific event tables that do not have a column named `user_actual_id`. This is a bit tricky to implement. A possible proxy is to implement a macro that iterates over the tables in the schema and gets the ones with both a `user_id` and `user_actual_id` column.  This can be a bit challenging to implement, and might grow the size of the current PR a bit.

Reply:
Also note that these synthetic tables can help answering questions like daily count of events per server/user etc. This seems to be a recurring pattern from product.

Reply:
"the goal was to keep only user generated events, and not server generated" - might be a good q to ask ourselves - what semantics are we missing (if any) by excluding the system-generated ones - like is the line between the two sets blurry or not? Could we potentially (now or in a future PR) tag them as such (like an `is_system` flag)  to be able to filter them out/build metrics targeting them specifically.